### ISSUE TITLE
Backspace doesn't work in xterm

### ISSUE THREAD

**Issue by aeosynth** [elves/elvish#328]
i was working with `xterm` today, and pressing backspace in elvish gave the error `Unbound: Ctrl-H`.

**Comment by huiyiqun** [comment]
You could do that by yourself with following in your rc.elv:

```
le:binding[insert][Ctrl+H] = &le:binding[insert][Backspace]
```

what's more, you could also check the `$E:TERM` variable before you do that.

**Comment by aeosynth** [comment]
i could do it myself, but the elvish tagline is "A friendly and expressive Unix shell". a broken backspace key is not friendly.

**Comment by aeosynth** [comment]
apparently xterm has an option which fixes this (ctrl+left-click, `Backarrow Key`)

**Comment by aeosynth** [comment]
reopening - the xterm `Backarrow Key` option fixes backspace in elvish, but breaks backspace in vi. another issue, `Alt+,`, `Alt+.` do not work; they insert these characters instead: `¬®`. if xterm is not supported, that should at least be documented.

**Comment by xiaq** [comment]
Xterm is quite different in terms of the escape sequences it sends. Detecting `$TERM` does not help, since most modern terminals also advertise themselves as `xterm`.

I don't know a good way to fix this yet, other than documenting it (0bd348a6fb1895f8a941afc6d81b87a31affe8d0). I shall leave it open for now.

**Comment by aeosynth** [comment]
backspace not working is the most critical issue, which could be trivially fixed by adding the ^H readline keybinding: #339
